lib/SILOptimizer/SILCombiner/SILCombinerMiscVisitors.cpp

```diff
@@ -91,8 +91,16 @@ SILCombiner::visitAllocExistentialBoxInst(AllocExistentialBoxInst *AEBI) {
   if (SingleStore && SingleRelease) {
     assert(SingleProjection && "store without a projection");
     // Release the value that was stored into the existential box. The box
-    // is going away so we need to release the stored value now.
-    Builder.setInsertionPoint(SingleStore);
+    // is going away so we need to release the stored value.
+    // Note that it's important that the release is inserted at the single
+    // release of the box and not at the store, because a balancing retain could
+    // be _after_ the store, e.g.
```

Comment:
You should be a colon after this. Also can you make Note in all capital letters? Something like this:

Release the value that was stored into the existential box. The box
is going away, so we need to release the stored value.

NOTE: It is important that the release is inserted at the single release of the box and /not/ at the store since a balancing retain could be _after_ the store, e.g.:

Reply:
thanks, I'll change it